content/terraform/v1.12.x/docs/language/resources/provisioners.mdx

```diff
@@ -0,0 +1,422 @@
+SSH or WinRM.
+
+> **Hands-on:** Try the [Provision Infrastructure with Packer](/terraform/tutorials/provision/packer?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.
+
+## Run the CLI
+
+The provider may support executing CLI commands on the resource. Refer to the resource provider documentation for instructions on how to invoke the CLI on your target system. If the provider does not include a mechanism for interacting with the resource CLI, we recommend opening a feature request ticket with the provider. 
+
+As a workaround, you can use a Terraform provisioner to run commands on the machine where Terraform or on a remote resource. Refer to [Configure a provisioner](#configure-a-provisioner) for instructions on adding provisioners to the resource configuration.
+
+### Commands on the local machine
+
+To run commands on the machine where Terraform is installed, add the `provisioner "local-exec"` block to your `resource` block and specify commands to run in the `command` argument. The following example configures an `aws_instance` resource named `web` to run a local command that prints its IP address:
+
+```hcl
+resource "aws_instance" "web" {
+  # ...
+
+  provisioner "local-exec" {
+    command = "echo The server's IP address is ${self.private_ip}"
+  }
+}
+```
+
+Refer to the [`provisioner` block reference](/terraform/language/resource#provisioner) for details about the available arguments.
+
+### Commands on a remote resource 
+
+To run commands on a remote resource, add the `remote-exec` provisioner type. This provisioner lets you run the CLI for your target system in order to create, update, or otherwise interact with remote objects in that system. You must include [a `connection` block](/terraform/language/resources/provisioners#connect-to-remote-resources) for Terraform to connect to the server. 
+
+
+## Execute remote scripts
+
+ If the provider does not include a mechanism for executing scripts, we recommend opening a feature request ticket with the provider. 
+
+As a workaround, you can use a Terraform provisioner to execute scripts. Refer to [Use a provisioner](#Use-a-provisioner) for instructions on adding provisioners to the resource configuration. 
+
+When provisioners execute scripts on a remote system over SSH, they usually upload the script file to the remote system before asking the default shell to execute it. Provisioners use this strategy because it allows you to use all of the typical scripting techniques supported by that shell, including preserving environment variable values and other context between script statements.
```

Comment:
```suggestion
When provisioners execute scripts on a remote system over SSH, they usually upload the script file to the remote system then use the default shell to execute it. Provisioners use this strategy because it allows you to use all of the typical scripting techniques supported by that shell, including preserving environment variable values and other context between script statements.
```